### PR TITLE
[BEAM-14386] [Flink] Support for scala 2.12

### PR DESCRIPTION
**Pull request by Miuler**
feat(flink/BEAM-14386): Use flink for scala 2.12

Reviewed-by: @angoenka @mxm @tweise 
Refs: # (pull request) [[BEAM-14386](https://issues.apache.org/jira/browse/BEAM-14386)] (jira)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [X] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [X] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [X] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Miuler** [comment]
This is a update from PR #17331, I'm create the jira issue https://issues.apache.org/jira/browse/BEAM-14386

**Comment by Miuler** [comment]
@ibzib this is the PR from feature to master (https://github.com/apache/beam/pull/17331)

**Comment by ibzib** [comment]
Can we add a note to CHANGES.md so users of the next Beam version can be aware of this change? https://github.com/apache/beam/blob/master/CHANGES.md

**Comment by Miuler** [comment]
> Can we add a note to CHANGES.md so users of the next Beam version can be aware of this change? https://github.com/apache/beam/blob/master/CHANGES.md


Yes, I do appended in the CHANGES.md
image

**Comment by Miuler** [comment]
> > Can we add a note to CHANGES.md so users of the next Beam version can be aware of this change? https://github.com/apache/beam/blob/master/CHANGES.md
> 
> @ibzib Yes, I do appended in the CHANGES.md image

mmm... should it be another message?

**Comment by Miuler** [comment]
The scala version 2.11 is deprecated in Flink and many more libraries for scala, the minimum version should by 2.12.

image

File: CHANGES.md [hunk 1]
```diff
@@ -81,6 +81,7 @@
 
 ## Deprecations
 
+* Support for flink scala 2.12, because most of the libraries support version 2.12 onwards. ([beam-14386](https://issues.apache.org/jira/browse/BEAM-14386))
```
Comment (y1chi) [inline CHANGES.md:84]: Should this go to breaking changes?

Comment (Miuler) [reply CHANGES.md:84]: I already moved it to the Breaking section

Comment (y1chi) [reply CHANGES.md:84]: there was no change from your last commit?

**Pull request merged**